### ISSUE TITLE
Drupal sends the number of active users per minute 

### ISSUE THREAD

**Issue by oksana-c** [department-of-veterans-affairs/va.gov-cms#4904]
## User Story

As a CMS Engineer, I need to send the number of active users per minute over time from Drupal to Prometheus, so I can collect and display data over time.

Active user = user is actively performing actions in the CMS, e.g. browsing, saving nodes, etc.

## Acceptance Criteria
- [x] Drupal sends the number of users that have a last access timestamp in the last ~60~ 180 seconds to Prometheus
- [x] CMS Application dashboard is created in Grafana and [shows a graph of active users per minute](http://grafana.vfs.va.gov/d/dxf8a-6Zz/cms-dashboard?viewPanel=47&orgId=1)
- [x] Prometheus collects info about logged in users.

## Implementation notes

- query `access` column in `users_field_data` table to find the number of users that have last access time in the 60 seconds. Current limitation of last access value is that it updates on every access, but no more than once every 180 seconds

https://www.drupal.org/project/drupal/issues/1835496
https://api.drupal.org/api/drupal/core%21modules%21user%21src%21EventSubscriber%21UserRequestSubscriber.php/function/UserRequestSubscriber%3A%3AgetSubscribedEvents/8.2.x

**Comment by mchelen-gov** [comment]
let's implement this in prometheus_exporter and it can be last 24 hrs

**Comment by acrollet** [comment]
@mchelen-gov can you clarify on last 24 hours, do you mean recording the number of users active in the last 24 hours? If so, that would work for reporting purposes but doesn't give very good granularity for scaling decisions based on peak usage periods... thanks!

**Comment by mchelen-gov** [comment]
@acrollet ah ok i see, are we trying to get insight into peak simultaneous user count?

**Comment by acrollet** [comment]
@mchelen-gov yeah, that was the motivation behind short check intervals :-)

**Comment by ndouglas** [comment]
We should discuss this in 16th minute today.

**TL;DR**: 
```php
  /**
   * Updates the current user's last access time.
   *
   * @param \Symfony\Component\HttpKernel\Event\TerminateEvent $event
   *   The event to process.
   */
  public function onKernelTerminate(TerminateEvent $event) {
    if ($this->account->isAuthenticated() && REQUEST_TIME - $this->account->getLastAccessedTime() > Settings::get('session_write_interval', 180)) {
      // Do that no more than once per 180 seconds.
      /** @var \Drupal\user\UserStorageInterface $storage */
      $storage = $this->entityTypeManager->getStorage('user');
      $storage->updateLastAccessTimestamp($this->account, REQUEST_TIME);
    }
  }
```

Suggestions:
- Change the requirement to checking users active within the last three minutes.  I'd prefer this, as it's the least likely to have a deleterious effect on performance and I don't think that users-in-the-past-one-minute represents a dramatic improvement over three minutes.
- Change the `session_write_interval` setting to 60.  Seems to be used in only a couple places in Core, but will have some performance impact proportional to the number of active users and the number of actions they perform.  IDK how substantial this is.  But it's mechanically easy to accomplish.
- Muck around with the data to estimate users active in the last minute.  Meh.

**Comment by mchelen-gov** [comment]
where is the Grafana dashboard?

**Comment by ndouglas** [comment]
@mchelen-gov Oops, thought that was in a separate issue.  I started a panel for this the other day but it hadn't deployed yet.

**Comment by ndouglas** [comment]
<img width="855" alt="Screen Shot 2021-09-17 at 12 26 51 PM" src="https://user-images.githubusercontent.com/1318579/133822264-1db6873f-0a99-4f72-8564-f8d105be70bc.png">

**Comment by ndouglas** [comment]
That's actually my first Grafana panel that I've directly authored, @mchelen-gov, pls be gentle

**Comment by olivereri** [comment]
> @mchelen-gov Oops, thought that was in a separate issue. I started a panel for this the other day but the commit hadn't deployed yet.

You're right, this was defined in #5418. Just had to add Total Users to meet the AC.

**Comment by mchelen-gov** [comment]
@ndouglas can you add links to AC?

**Comment by ndouglas** [comment]
@mchelen-gov done 😃
